WordPress/Classes/ViewRelated/Blog/BlogListViewController.m

```diff
@@ -82,11 +83,15 @@ - (void)configureNavigationBar
     self.addSiteButton = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemAdd
                                                                                    target:self
                                                                                    action:@selector(addSite)];
-    UIBarButtonItem *searchButton = [[UIBarButtonItem alloc] initWithImage:[UIImage imageNamed:@"icon-post-search"]
-                                                                     style:UIBarButtonItemStylePlain
-                                                                    target:self
-                                                                    action:@selector(toggleSearch)];
-    [self.navigationItem setRightBarButtonItems:@[self.addSiteButton, searchButton]];
+    
+    self.searchButton = [[UIBarButtonItem alloc] initWithImage:[UIImage imageNamed:@"icon-post-search"]
+                                                         style:UIBarButtonItemStylePlain
+                                                        target:self
+                                                        action:@selector(toggleSearch)];
+
+    self.navigationItem.rightBarButtonItems = @[ self.addSiteButton, self.searchButton ];
```

Comment:
instead of setting `rightBarButtonItems` here, how about calling `[self updateSearchButton]`? This way it'll init with the correct version.